### ISSUE TITLE
Get-AzureRmADAppCredential with provided ApplicationId throws a exception

### ISSUE THREAD

**Issue by leowumsft** [Azure/azure-powershell#4741]
### PowerShell Version
5.1.15063.608

### Module Version
4.4.0

### OS Version
10.0.15063.608

### Description
Execute the cmdlet Get-AzureRmADAppCredential by providing ApplicationId parameter.   ApplicationId can be a valid one or a dummy value.   It always output the exception.

### Debug Output

```
DEBUG: 10:35:13 AM - GetAzureADAppCredentialCommand begin processing with ParameterSet 'ApplicationIdParameterSet'.
DEBUG: 10:35:13 AM - using account id '***************'...
DEBUG: [Common.Authentication]: Authenticating using Account: '*****************', environment:
'AzureUSGovernment', tenant: 'ca143b43-8298-4111-9205-3e2676af874b'
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 959b2617-16f7-4fea-94a5-bb915844b031 - AcquireTokenHandlerBase: === Token Acquisition
started:
 Authority: https://login-us.microsoftonline.com/ca143b43-8298-4111-9205-3e2676af874b/
 Resource: https://graph.windows.net/
 ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
 CacheType: Microsoft.Azure.Commands.Common.Authentication.AuthenticationStoreTokenCache (16 items)
 Authentication Target: User

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 10/09/2017 14:35:13: 959b2617-16f7-4fea-94a5-bb915844b031 - TokenCache: Looking up cache for a token...
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 959b2617-16f7-4fea-94a5-bb915844b031 - TokenCache: An item matching the requested resource
was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 10/09/2017 14:35:13: 959b2617-16f7-4fea-94a5-bb915844b031 - TokenCache: 50.9029326366667 minutes left until
token in cache expires
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 959b2617-16f7-4fea-94a5-bb915844b031 - TokenCache: A matching item (access token or refresh
 token or both) was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 959b2617-16f7-4fea-94a5-bb915844b031 - AcquireTokenHandlerBase: === Token Acquisition
finished successfully. An access token was retuned:
 Access Token Hash: R1fj9Pd944IIJSv97eu5EpgIcL5UqISxGU2oMjoxf8Q=
 Refresh Token Hash: OUHs+jI29YSU6P/wisAZe/mYZpeHg2Zu0hqLb+2IgcI=
 Expiration Time: 10/09/2017 15:26:07 +00:00
 User Hash: z/31PDfsDxScJwHSY3d1BuX/AA6E98uibsV8Wcbvyjo=

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13:  - TokenCache: Serializing token cache with 16 items.
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 2747dbe4-7cf0-4158-bcd4-75b0c79ed893 - AcquireTokenHandlerBase: === Token Acquisition
started:
 Authority: https://login-us.microsoftonline.com/ca143b43-8298-4111-9205-3e2676af874b/
 Resource: https://graph.windows.net/
 ClientId: 1950a258-227b-4e31-a9cf-717495945fc2
 CacheType: Microsoft.Azure.Commands.Common.Authentication.AuthenticationStoreTokenCache (16 items)
 Authentication Target: User

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 10/09/2017 14:35:13: 2747dbe4-7cf0-4158-bcd4-75b0c79ed893 - TokenCache: Looking up cache for a token...
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 2747dbe4-7cf0-4158-bcd4-75b0c79ed893 - TokenCache: An item matching the requested resource
was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Verbose: 1 :
DEBUG: 10/09/2017 14:35:13: 2747dbe4-7cf0-4158-bcd4-75b0c79ed893 - TokenCache: 50.90284932 minutes left until token in
cache expires
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 2747dbe4-7cf0-4158-bcd4-75b0c79ed893 - TokenCache: A matching item (access token or refresh
 token or both) was found in the cache
DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13: 2747dbe4-7cf0-4158-bcd4-75b0c79ed893 - AcquireTokenHandlerBase: === Token Acquisition
finished successfully. An access token was retuned:
 Access Token Hash: R1fj9Pd944IIJSv97eu5EpgIcL5UqISxGU2oMjoxf8Q=
 Refresh Token Hash: OUHs+jI29YSU6P/wisAZe/mYZpeHg2Zu0hqLb+2IgcI=
 Expiration Time: 10/09/2017 15:26:07 +00:00
 User Hash: z/31PDfsDxScJwHSY3d1BuX/AA6E98uibsV8Wcbvyjo=

DEBUG: Microsoft.IdentityModel.Clients.ActiveDirectory Information: 2 :
DEBUG: 10/09/2017 14:35:13:  - TokenCache: Serializing token cache with 16 items.
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
GET

Absolute Uri:
https://graph.windows.net/ca143b43-8298-4111-9205-3e2676af874b/applications?$filter=appId eq
'asfsdasdfasdf'&api-version=1.6

Headers:
x-ms-client-request-id        : ff4cf01d-b1a7-4aa0-9b2b-ceba4619ee94
accept-language               : en-US

Body:


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
ocp-aad-diagnostics-server-name: csudcxj3KFkmOZf31RQWEPIy+W22QvY4rjm0SCwVaEQ=
request-id                    : aaf4283b-f8a8-4ebb-be01-4763a6a73b91
client-request-id             : a93304ad-8f4f-4726-bb54-b077ff1b7839
x-ms-dirapi-data-contract-version: 1.6
ocp-aad-session-key           :
JAi22Hwe_eZHlV1F0Q-8WrzehtElvZMg2hxW6UZSEah3E8IQ7uKFRmhF-b1HAr1h70l97jP4krpui6i9gAMhL2KliXetkdt3tyYHG3vPx6EBkDGkuHItm6m
ylaXjAZDa.wmLmvr6FmE_Bj-aaL4Bd8a6O-0eW7yCiAQdow5flmvg
Strict-Transport-Security     : max-age=31536000; includeSubDomains
Access-Control-Allow-Origin   : *
Duration                      : 1191650
Cache-Control                 : private
Server                        : Microsoft-IIS/8.5
X-AspNet-Version              : 4.0.30319
X-Powered-By                  : ASP.NET,ASP.NET
Date                          : Mon, 09 Oct 2017 14:35:12 GMT

Body:
{
  "odata.error": {
    "code": "Request_UnsupportedQuery",
    "message": {
      "lang": "en",
      "value": "Unsupported or invalid query filter clause specified for property 'appId' of resource 'Application'."
    }
  }
}

DEBUG: Caught exception, type: Microsoft.Azure.Graph.RBAC.Version1_6.Models.GraphErrorException
Get-AzureRmADAppCredential : Unsupported or invalid query filter clause specified for property 'appId' of resource
'Application'.
At line:1 char:1
+ Get-AzureRmADAppCredential -ApplicationId asfsdasdfasdf
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Get-AzureRmADAppCredential], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ActiveDirectory.GetAzureADAppCredentialCommand

DEBUG: 10:35:13 AM - GetAzureADAppCredentialCommand end processing.
DEBUG: 10:35:13 AM - GetAzureADAppCredentialCommand end processing.
```

### Script/Steps for Reproduction
Get-AzureRmADAppCredential -ApplicationId asfsdasdfasdf

**Comment by cormacpayne** [comment]
@darshanhs90 Hey Haridarshan, would you mind taking a look at this issue?

**Comment by cormacpayne** [comment]
@leowumsft Hey Leo, I am unable to reproduce this issue when providing a _valid_ value for `ApplicationId`, but I do see this issue when providing a dummy value. I'll keep this issue open to track the status of updating the error message that is thrown from this.

**Comment by cormacpayne** [comment]
@darshanhs90 Hey Haridarshan, it looks like an incorrect error is thrown when a non-Guid string is provided to `ApplicationId`:

```powershell
Get-AzureRmADApplication -ApplicationId ffffffff-ffff-ffff-ffff-ffffffffffff

Get-AzureRmADAppCredential : Application with AppId 'ffffffff-ffff-ffff-ffff-ffffffffffff' does not exist.
At line:1 char:1
+ Get-AzureRmADAppCredential -ApplicationId ffffffff-ffff-ffff-ffff-fff ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Get-AzureRmADAppCredential], InvalidOperationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ActiveDirectory.GetAzureADAppCredentialCommand


Get-AzureRmADApplication -ApplicationId asdf

Get-AzureRmADAppCredential : Unsupported or invalid query filter clause specified for property 'appId' of resource 'Application'.
At line:1 char:1
+ Get-AzureRmADAppCredential -ApplicationId asdf
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (:) [Get-AzureRmADAppCredential], Exception
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ActiveDirectory.GetAzureADAppCredentialCommand
```

I think the correct approach here would be changing the type of `ApplicationId` from a string to a Guid, preventing users from providing an incorrectly formatted string.

**Comment by cormacpayne** [comment]
@leowumsft @darshanhs90 I fixed this issue in my development branch for the AD cmdlets that we are looking to get into Azure PowerShell for the May release. I made the [`ApplicationId`](https://github.com/cormacpayne/azure-powershell/blob/resources-improvements/src/ResourceManager/Resources/Commands.Resources/help/Get-AzureRmADAppCredential.md#-applicationid) parameter accept a `Guid` instead of a `string`, which will prevent users from seeing the above filter error.

**Comment by cormacpayne** [comment]
This will be fixed in version 6.0.0 of `AzureRM` (version 6.0.0 of `AzureRM.Resources`) that is released on May 7th. Please feel free to re-open this issue if the problem persists.
